File: 02_automatedSimplifiedExtensionLayer/00_matlab/functions/work_done_Hv.m
function [progress, flow_names_missing_Hv]=work_done_Hv(Model)
% calculate how many flows where already assessed by words

work_to_do=0;
flow_amounts=size(Model.meta_data_flows(2:end,4),1);
counter=0;
flow_names_missing_Hv={[]};
for i=2:size(Model.meta_data_flows,1)
    if isempty(Model.meta_data_flows{i,9})
       work_to_do=work_to_do+1;
       counter=counter+1;
       flow_names_missing_Hv(counter)=Model.meta_data_flows(i,1);
    end
end
flow_names_missing_Hv=flow_names_missing_Hv';
progress=(flow_amounts-work_to_do)/flow_amounts;

File: 02_automatedSimplifiedExtensionLayer/utils/work_done_Hv.py
def work_done_Hv(Model):
    # Calculate how many flows were already assessed by words

    work_to_do = 0  # Initialize counter for work still to be done
    flow_amounts = len(Model['meta_data_flows'][1:])  # Number of flows excluding the header (row 1 in MATLAB)
    counter = 0  # Counter for missing flows
    flow_names_missing_Hv = []  # Initialize an empty list for missing Hv flows

    # Loop through all rows starting from index 1 to match MATLAB's i = 2 (skipping header)
    for i in range(1, len(Model['meta_data_flows'])):
        if not Model['meta_data_flows'][i][8]:  # Check if the 9th column (index 8) is empty
            work_to_do += 1
            counter += 1
            flow_names_missing_Hv.append(Model['meta_data_flows'][i][0])  # Add the flow name to the list

    flow_names_missing_Hv = list(flow_names_missing_Hv)  # Make sure it's a list
    progress = (flow_amounts - work_to_do) / flow_amounts  # Calculate the progress

    return progress, flow_names_missing_Hv
